panic on loading a tfstate file that contains a resource with an instances as an empty array

Hi! Thank you for a useful tool 😆

As mentioned in the title, I found that some of the tfstate files I have are not able to be looked up and are panicked.
After some investigation, I found out that this happens when the file contains a resource with an `instances` as an empty array.
The following are the minimum steps to reproduce it.

```shell
tfstate-lookup on 🔀 master is 📦 v0.1.1 
❯ cat reproduce.tfstate
{
  "version": 4,
  "terraform_version": "0.12.19",
  "serial": 1,
  "lineage": "00000000-0000-0000-0000-000000000000",
  "outputs": {},
  "resources": [
    {
      "module": "module.example",
      "mode": "managed",
      "type": "aws_vpc",
      "name": "example",
      "provider": "provider.aws",
      "instances": []
    }
  ]
}

❯ tfstate-lookup -s reproduce.tfstate module.example.aws_vpc.example
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/fujiwara/tfstate-lookup/tfstate.(*TFState).scan(0xc00017ebd0)
        /home/runner/work/tfstate-lookup/tfstate-lookup/tfstate/lookup.go:224 +0x9af
sync.(*Once).doSlow(0xc00017ec30, 0xc00051fcc8)
        /opt/hostedtoolcache/go/1.15.6/x64/src/sync/once.go:66 +0xec
sync.(*Once).Do(...)
        /opt/hostedtoolcache/go/1.15.6/x64/src/sync/once.go:57
github.com/fujiwara/tfstate-lookup/tfstate.(*TFState).Lookup(0xc00017ebd0, 0x7ffeefbff08c, 0x1e, 0x0, 0x0, 0x0)
        /home/runner/work/tfstate-lookup/tfstate-lookup/tfstate/lookup.go:177 +0x35a
main._main(0x1ad6700, 0xc000058778)
        /home/runner/work/tfstate-lookup/tfstate-lookup/cmd/tfstate-lookup/main.go:54 +0x3bc
main.main()
        /home/runner/work/tfstate-lookup/tfstate-lookup/cmd/tfstate-lookup/main.go:21 +0x26
```

I don't know when and why Terraform produces instances as an empty array to the file, but this kind of a file actually exists (I didn't edit it manually, of course), and I haven't had any problems doing normal Terraform operations such as `terraform plan` or `terraform apply` etc 😅 
This means that there may be other people besides me who encounter this problem.

By the way, for your reference, no error occurred in the following cases 🤔 

```shell
❯ cat with_each_list.tfstate
{
  "version": 4,
  "terraform_version": "0.12.19",
  "serial": 1,
  "lineage": "00000000-0000-0000-0000-000000000000",
  "outputs": {},
  "resources": [
    {
      "module": "module.example",
      "mode": "managed",
      "type": "aws_vpc",
      "name": "example",
      "each": "list",
      "provider": "provider.aws",
      "instances": []
    }
  ]
}

❯ tfstate-lookup -s with_each_list.tfstate module.example.aws_vpc.example
null

❯ cat with_empty_object.tfstate
{
  "version": 4,
  "terraform_version": "0.12.19",
  "serial": 1,
  "lineage": "00000000-0000-0000-0000-000000000000",
  "outputs": {},
  "resources": [
    {
      "module": "module.example",
      "mode": "managed",
      "type": "aws_vpc",
      "name": "example",
      "provider": "provider.aws",
      "instances": [{}]
    }
  ]
}

❯ tfstate-lookup -s with_empty_object.tfstate module.example.aws_vpc.example
null
```

Can you please take a look at it? Thanks in advance!

--- a/tfstate/lookup.go
+++ b/tfstate/lookup.go
@@ -221,8 +221,10 @@ func (s *TFState) scan() {
 				}
 			} else {
 				key := module + fmt.Sprintf("%s%s.%s", prefix, r.Type, r.Name)
-				ins := r.Instances[0]
-				s.scanned[key] = ins
+				if len(r.Instances) != 0 {
+					ins := r.Instances[0]
+					s.scanned[key] = ins
+				}
 			}
 		}
 	}

--- a/tfstate/lookup_test.go
+++ b/tfstate/lookup_test.go
@@ -141,6 +141,10 @@ var TestSuitesOK = []TestSuite{
 		Key:    `data.aws_lb_target_group.app["dev1"].name`,
 		Result: "dev-dev1-app",
 	},
+	{
+		Key:    `module.example.aws_vpc.example`,
+		Result: nil,
+	},
 }
 
 func testLookupState(t *testing.T, state *tfstate.TFState) {

--- a/tfstate/test/terraform.tfstate
+++ b/tfstate/test/terraform.tfstate
@@ -405,6 +405,14 @@
           }
         }
       ]
+    },
+    {
+      "module": "module.example",
+      "mode": "managed",
+      "type": "aws_vpc",
+      "name": "example",
+      "provider": "provider.aws",
+      "instances": []
     }
   ]
 }

